x-pack/plugins/apm/public/services/rest/savedObjects.js

```diff
@@ -10,21 +10,23 @@ import { getFromSavedObject } from 'ui/index_patterns/static_utils';
 import { callApi } from './callApi';
 
 export const getAPMIndexPattern = memoize(async () => {
+  const apmIndexPatternTitle = chrome.getInjected('apmIndexPatternTitle');
   const res = await callApi({
     pathname: `/api/saved_objects/_find`,
     query: {
-      type: 'index-pattern'
+      type: 'index-pattern',
+      search: `"${apmIndexPatternTitle}"`,
+      per_page: 200
```

Comment:
This doesn't actually paginate through all the pages but the narrowed-down search query combined with the increased per_page option should be sufficient. I can increase to 1000 if necessary but seemed a big excessive.